src/realm/sync/noinst/migration_store.cpp

```diff
@@ -53,14 +53,14 @@ bool MigrationStore::load_data(bool read_only)
 
     auto tr = m_db->start_read();
     // Start with a reader so it doesn't try to write until we are ready
-    SyncMetadataSchemaVersionsReader schema_versions_reader(tr);
+    SyncMetadataSchemaVersionsReader schema_versions_reader(*tr);
```

Comment:
aside from passing around a ref vs const ref to a shared_ptr, is there a reason why this changed from a Transaction to a Group?

Reply:
Aside from passing around a const ref instead of a const ref to a shared_ptr, is there a reason the parameter is a Group now and not a Transaction?

Reply:
aside from passing around a ref vs ptr, is there a reason why this changed from a Transaction to a Group?

Reply:
The root change is enabling `PendingResetStore::has_pending_reset(realm->read_group())`, which previously didn't work because the function expected a Transaction even though it didn't do anything which required a Transaction. All of these functions should have been taking a Group the whole time as they don't change the transaction state.